server/routes/user.js

```diff
@@ -112,6 +116,68 @@ router.post(
   }
 );
 
+//Get all notifications of a user Id
+router.get("/getAllNotifications/:id", async (req, res) => {
+  try {
+    const { id } = req.params;
+
+    //Through the UserNotification Model from below, we'll be able to obtain the
+    //notification_id that is only present on the particular user
+    const userNotification = await UserNotificationModel.findAll({
+      where: { user_id: id },
+    });
+
+    const promiseArr = [];
+
+    //Get all the notifications for each of the values in the array
+    userNotification.forEach((userNotif) => {
+      promiseArr.push(
+        NotificationModel.findOne({ where: { id: userNotif.notification_id } })
+      );
+    });
+
+    const data = await Promise.all(promiseArr);
+
+    res.send(data);
+  } catch (error) {
+    res.status(500).send({
+      msg: "Server Error",
+    });
+  }
+});
+
+//Get all unread using user id
+router.get("/getAllUnreadNotifications/:id", async (req, res) => {
+  try {
+    const { id } = req.params;
+
+    const userNotification = await UserNotificationModel.findAll({
+      where: { user_id: id },
+    });
+
+    let promiseArr = [];
+
+    userNotification.forEach((element) => {
+      promiseArr.push(
+        NotificationModel.findOne({
+          where: { id: element.notification_id, isRead: false },
+        })
+      );
+    });
+
+    let data = await Promise.all(promiseArr);
+
+    data = data[0] === null ? [] : data;
```

Comment:
The result of calling `NotificationModel.findOne({...})` and not finding a result is `null`?